addon/plugins/variable-plugin/utils/constants.ts

```diff
@@ -92,23 +94,25 @@ export const DEFAULT_VARIABLE_TYPES: Record<string, VariableType> = {
       const codelists = fetchCodeListsByPublisher(endpoint, publisher);
       return codelists;
     },
-    constructor: (schema, label, endpoint, selectedCodelist?: CodeList) => {
+    constructor: ({ schema, attributes, codelist, label }) => {
       const mappingURI = `http://data.lblod.info/mappings/${uuidv4()}`;
       const variableInstance = `http://data.lblod.info/variables/${uuidv4()}`;
       return schema.node(
         'variable',
         {
           type: 'codelist',
           mappingResource: mappingURI,
-          codelistResource: selectedCodelist?.uri,
+          codelistResource: codelist?.uri,
           variableInstance,
-          source: endpoint,
-          label: label ?? selectedCodelist?.label,
+          label: label ?? codelist?.label,
+          ...attributes,
         },
         schema.node('placeholder', {
-          placeholderText: selectedCodelist?.label,
+          placeholderText: codelist?.label,
         })
       );
     },
   },
 };
+
+export const EXTRA_ATTRIBUTES_KEY = 'extraAttributes';
```

Comment:
maybe the extra attributes should be written to the html in the data- attributes